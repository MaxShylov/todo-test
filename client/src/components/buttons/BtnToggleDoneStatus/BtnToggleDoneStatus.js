import React  from 'react';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Button from 'antd/lib/button/button';
import Tooltip from 'antd/lib/tooltip';
import { toggleDoneStatus } from '../../../redux/actions/tasks/tasks';

const { number, bool, func } = PropTypes;


BtnToggleDoneStatus.propTypes = {
  id: number.isRequired,
  isDone: bool,
  // redux
  toggleDoneStatus: func
};


export function BtnToggleDoneStatus(props) {

  const { toggleDoneStatus, id, isDone } = props;
  const toggleStatus = () => toggleDoneStatus(id);

  return (
    <Tooltip
      placement="top"
      title={`Mark as ${!isDone ? '' : 'not'} done`}
    >
      <Button
        type="primary"
        icon={!isDone ? 'clock-circle' : 'check'}
        onClick={toggleStatus}
      />
    </Tooltip>
  );
}


const matchDispatchToProps = dispatch => bindActionCreators({
  toggleDoneStatus
}, dispatch);

export default connect(null, matchDispatchToProps)(BtnToggleDoneStatus);
