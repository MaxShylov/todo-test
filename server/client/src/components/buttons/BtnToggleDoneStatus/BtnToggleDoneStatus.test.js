import React from 'react';
import { shallow } from 'enzyme'

import { BtnToggleDoneStatus } from './BtnToggleDoneStatus.js'


describe('BtnToggleDoneStatus', () => {

  const props = {
    id: 1
  };

  describe('initial', () => {
    const wrapper = shallow(<BtnToggleDoneStatus {...props} />);

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
  });

  describe('function toggleStatus', () => {
    const
      mockToggleDoneStatus = jest.fn(),
      nextProps = {
        ...props,
        toggleDoneStatus: mockToggleDoneStatus
      },
      wrapper = shallow(<BtnToggleDoneStatus {...nextProps} />);

    wrapper.find('Button').simulate('click');

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('toggleDoneStatus called', () => expect(mockToggleDoneStatus).toHaveBeenCalled());
  });

});
