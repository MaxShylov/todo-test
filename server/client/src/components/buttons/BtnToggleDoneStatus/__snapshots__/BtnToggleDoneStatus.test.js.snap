// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnToggleDoneStatus function toggleStatus renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BtnToggleDoneStatus
    id={1}
    toggleDoneStatus={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "arrowPointAtCenter": false,
      "autoAdjustOverflow": true,
      "children": <Button
        block={false}
        ghost={false}
        icon="clock-circle"
        loading={false}
        onClick={[Function]}
        type="primary"
      />,
      "mouseEnterDelay": 0.1,
      "mouseLeaveDelay": 0.1,
      "placement": "top",
      "title": "Mark as  done",
      "transitionName": "zoom-big-fast",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "block": false,
        "ghost": false,
        "icon": "clock-circle",
        "loading": false,
        "onClick": [Function],
        "type": "primary",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "arrowPointAtCenter": false,
        "autoAdjustOverflow": true,
        "children": <Button
          block={false}
          ghost={false}
          icon="clock-circle"
          loading={false}
          onClick={[Function]}
          type="primary"
        />,
        "mouseEnterDelay": 0.1,
        "mouseLeaveDelay": 0.1,
        "placement": "top",
        "title": "Mark as  done",
        "transitionName": "zoom-big-fast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "ghost": false,
          "icon": "clock-circle",
          "loading": false,
          "onClick": [Function],
          "type": "primary",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`BtnToggleDoneStatus initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BtnToggleDoneStatus
    id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "arrowPointAtCenter": false,
      "autoAdjustOverflow": true,
      "children": <Button
        block={false}
        ghost={false}
        icon="clock-circle"
        loading={false}
        onClick={[Function]}
        type="primary"
      />,
      "mouseEnterDelay": 0.1,
      "mouseLeaveDelay": 0.1,
      "placement": "top",
      "title": "Mark as  done",
      "transitionName": "zoom-big-fast",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "block": false,
        "ghost": false,
        "icon": "clock-circle",
        "loading": false,
        "onClick": [Function],
        "type": "primary",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "arrowPointAtCenter": false,
        "autoAdjustOverflow": true,
        "children": <Button
          block={false}
          ghost={false}
          icon="clock-circle"
          loading={false}
          onClick={[Function]}
          type="primary"
        />,
        "mouseEnterDelay": 0.1,
        "mouseLeaveDelay": 0.1,
        "placement": "top",
        "title": "Mark as  done",
        "transitionName": "zoom-big-fast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "ghost": false,
          "icon": "clock-circle",
          "loading": false,
          "onClick": [Function],
          "type": "primary",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
